libraries/standard/mqtt/include/types/iot_mqtt_types.h

```diff
@@ -685,17 +685,135 @@ typedef struct IotMqttConnectInfo
     uint16_t passwordLength; /**< @brief Length of #IotMqttConnectInfo_t.pPassword. */
 } IotMqttConnectInfo_t;
 
-#if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
-
 /**
  * @cond DOXYGEN_IGNORE
  * Doxygen should ignore this section.
  *
  * Forward declaration of the internal MQTT packet structure.
  */
-    struct _mqttPacket;
+struct _mqttPacket;
 /** @endcond */
 
+/**
+ * @brief Get the MQTT packet type from a stream of bytes off the network.
+ *
+ * @param[in] pNetworkConnection Reference to the network connection.
+ * @param[in] pNetworkInterface Function pointers used to interact with the
+ * network.
+ */
+typedef uint8_t ( * IotMqttGetPacketTypeFunc_t )(
+                     void * pNetworkConnection,
+                     const IotNetworkInterface_t * pNetworkInterface );
+
+/**
+ * @brief Get the remaining length from a stream of bytes off the network.
+ *
+ * @param[in] pNetworkConnection Reference to the network connection.
+ * @param[in] pNetworkInterface Function pointers used to interact with the
+ * network.
+ */
+typedef size_t ( * IotMqttGetRemainingLengthFunc_t )(
+                     void * pNetworkConnection,
+                     const IotNetworkInterface_t * pNetworkInterface );
+
+/**
+ * @brief Free a packet generated by the serializer.
+ *
+ * This function pointer must be set if any other serializer override is set.
+ * @param[in] uint8_t* The packet to free.
+ */
+typedef void ( * IotMqttFreePacketFunc_t )( uint8_t * pPacket );
+
+/**
+ * @brief CONNECT packet serializer function.
+ * @param[in] IotMqttConnectInfo_t* User-provided CONNECT information.
+ * @param[out] uint8_t** Where the CONNECT packet is written.
+ * @param[out] size_t* Size of the CONNECT packet.
+ */
+typedef IotMqttError_t ( * IotMqttConnectSerializer_t )(
```

Comment:
Nit: The default implementations of these functions are called `_IotMqtt_SerializeConnect`, `_IotMqtt_SerializePingreq`, etc.

For consistency, could you rename these `IotMqttSerializeConnect_t`, `IotMqttSerializePingreq_t`, etc.

Reply:
I chose `Serializer` to make it clear that objects of this type are functions, but those names could be clearer, will change.